File: harness/make_worksheet.py
#!/usr/bin/env python3
"""Scaffold a human labeling worksheet (CSV) from a candidates file.

One row per (patient, engine-eligible HCC), with M.E.A.T. anchor columns the
reviewer fills to decide gold_status. Convert a filled sheet back to gold-label
JSONL with --to-gold.

Examples:
  make_worksheet.py --candidates candidates.jsonl --out worksheet.csv
  make_worksheet.py --to-gold worksheet.csv --out gold_labels.jsonl
"""
import argparse
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parent))
from hcceval import schema  # noqa: E402
from hcceval.crosswalk import DEFAULT_CROSSWALK, DEFAULT_HIERARCHY, Crosswalk  # noqa: E402
from hcceval.worksheet import build_worksheet, worksheet_to_gold  # noqa: E402


def main() -> int:
    p = argparse.ArgumentParser(description=__doc__)
    p.add_argument("--candidates", help="candidates JSONL (build a blank sheet)")
    p.add_argument("--to-gold", metavar="WORKSHEET_CSV",
                   help="convert a filled worksheet back to gold-label JSONL")
    p.add_argument("--crosswalk", default=str(DEFAULT_CROSSWALK))
    p.add_argument("--out", required=True, help="output path (CSV or JSONL)")
    args = p.parse_args()

    if args.to_gold:
        gold = worksheet_to_gold(args.to_gold)
        schema.write_jsonl(args.out, gold)
        print(f"wrote {len(gold)} gold labels -> {args.out}", file=sys.stderr)
        return 0

    if not args.candidates:
        p.error("provide --candidates (to build a sheet) or --to-gold (to convert one)")
    xwalk = Crosswalk.load(args.crosswalk, DEFAULT_HIERARCHY)
    cands = schema.load_candidates(args.candidates)
    n = build_worksheet(cands, xwalk, args.out)
    print(f"wrote worksheet with {n} rows -> {args.out}", file=sys.stderr)
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: crates/eval/src/bin/make_worksheet.rs
//! Scaffold a labeling worksheet from candidates, or convert a filled one to gold.
//!
//! Usage:
//!   make_worksheet --candidates C.jsonl --out worksheet.csv [--crosswalk CSV]
//!   make_worksheet --to-gold worksheet.csv --out gold.jsonl

use agent::crosswalk::Crosswalk;
use eval::{schema, worksheet};

fn arg(a: &[String], k: &str) -> Option<String> {
    a.iter().position(|x| x == k).and_then(|i| a.get(i + 1)).cloned()
}

fn main() {
    let a: Vec<String> = std::env::args().collect();
    let out = arg(&a, "--out").unwrap_or_else(|| {
        eprintln!("usage: make_worksheet --candidates C.jsonl --out worksheet.csv | --to-gold worksheet.csv --out gold.jsonl");
        std::process::exit(2);
    });

    if let Some(sheet) = arg(&a, "--to-gold") {
        let gold = worksheet::worksheet_to_gold(&sheet).expect("parse worksheet");
        let n = schema::write_jsonl(&out, &gold).expect("write gold");
        println!("converted {n} labeled rows -> {out}");
    } else if let Some(cand_p) = arg(&a, "--candidates") {
        let candidates = schema::load_candidates(&cand_p).expect("load candidates");
        let xwalk_p = arg(&a, "--crosswalk").unwrap_or_else(|| "crosswalks/snomed_to_icd10_v28.csv".into());
        let xwalk = Crosswalk::load(&xwalk_p).expect("load crosswalk");
        let n = worksheet::build_worksheet(&candidates, &xwalk, &out).expect("write worksheet");
        println!("wrote worksheet with {n} rows -> {out}");
    } else {
        eprintln!("need --candidates (build) or --to-gold (convert)");
        std::process::exit(2);
    }
}
